spark-publish/index.js

```diff
@@ -65,7 +65,14 @@ export const publish = async ({
   // Call contract with CID
   const signal = AbortSignal.timeout(600_000) // 10-minute timeout
   const { roundIndex, ieAddMeasurementsDuration } = await pTimeout(
-    commitMeasurements({ cid, ieContract, logger, signal }),
+    pRetry(
+      () => commitMeasurements({ cid, ieContract, logger, signal }),
+      {
+        onFailedAttempt: err => console.error(err),
```

Comment:
I think this will log some errors twice: the first time from `onFailedAttempt` configured for `pRetry(tx.wait)` and the second time from this place. However, errors thrown outside of pRetry will not be logged that way and we still need to log them here.

It would be great if we could detect the case when an error was already observed via the inner `onFailedAttempt` handler. I am not sure if we can use one of the additional error properties added by pRetry, e.g. `err.attemptNumber` or `err.retriesLeft` or whether we would have to add our own property in the `onFailedAttempt` handler.

Anyhow, this is a minor detail we can ignore for now and wait if it ever becomes a problem in production.

Reply:
This only happens when the last attempt of the inner `p-retry` is exhausted, right?

Reply:
I'm going to merge for now, let's continue discussion here 🙏